hi-ml-histopathology/testhisto/testhisto/datamodules/test_slides_datamodule.py

```diff
@@ -0,0 +1,107 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import py
+import pytest
+import numpy as np
+
+from health_ml.utils.common_utils import is_gpu_available
+from testhisto.utils.utils_slides_datamodule import MockSlidesDataModule, MockWSIGenerator, MockWSIType
+
+
+no_gpu = not is_gpu_available()
+N_REPEAT_DIAG = 4
+
+
+@pytest.fixture(scope="session")
+def mock_wsi_root_dir(tmpdir_factory: pytest.TempdirFactory) -> py.path.local:
+    tmp_root_dir = tmpdir_factory.mktemp("mock_wsi")
```

Comment:
pytest has a fixture called "tmp_path" that you can use here, I think that's more easy than the tmpdirfactory

Reply:
I used the tmpdirfactory because it is a session-scoped fixture so that we generate the mock data once for the entire test suite.
Here's the [doc](https://docs.pytest.org/en/6.2.x/tmpdir.html). Or do you prefer the tmp_path fixture?

Reply:
ah, yes, that makes sense. Leave as is.